designs/virtualization/4.6/Next-run-configuration/Next-run-configuration.md

```diff
@@ -0,0 +1,79 @@
+---
+parent: Virtualization
+version: 4.6
+---
+# Next run configuration
+
+As a user makes changes to a running Virtual machine they will need to wait for the VM to restart to see the changes applied (the changes won’t be applied until the next start of the VM).
+
+## Editing
+
+When a user clicks to edit/ update one of the attributes (f.e. Flavor) it will open a modal with a notification letting the user know that the VM must be restarted in order to see the changes applied. The notification will persist until they restart the VM.
+This notification will not appear until the user has made changes, so the user will then know that they will need to restart.
+
+In general, users will be able to clear the changes and keep what they had, but when the first time the modal is opened, in case the user hasn't saved any changes yet, the 'Clear changes' button will not be shown.
+
+To avoid confusion users will be able to view the current settings in The VM Details page.
+
+![user is editing values](img/EditOriginal.png)
+
+At this point the user can either just save the requested changes and postpone the restart or save and restart the VM.
+
+## Restart modal
+
+If the user chooses to save and restart the VM they will be asked again to ensure that and will get into a modal including an inline notification that restarting may be disruptive to some logged in users and a notification that the new setting will be applied upon restart. The notification will include a confirmation checkbox so they would be aware of the effect that the restart action might cause and a link to the logged in users list at the bottom of the VM Details page.
+
+![user chooses to resart the VM](img/RestartModal.png)
+
+## VM Details page
+
+In the VM Details page they will be notified that the changes they have made will get applied only after the VM will be restarted.
+In the ‘Flavor’ section, beside the Edit icon, we will add (pending changes) that will lead to the Edit Flavor modal where they can view the pending changes.
+Only in cases when users have made changes to the part of the YAML which are not exposed in the ui, we will add a link to ‘View pending changes in YAML editor’.
+
+![VM details page](img/vm-details.png)
+
+In case there are multiple changes that are on different tabs, we will show a list of them and the same notification will be shown on every tab. Clicking each will take the user to the relevant tab and will open the modal.
+An environment change can also be presented as a Disk change (if a user adds an environment key-value it is being added as a disk), so a change in the environment tab should be reflected in both 'Disks' and 'Environment' tabs and include a separate inline notification about the changes with the ability to clear them.
+
+For changes that apply across all tabs, we'll place the notification above the tabs, so users will always be able to see it.
+
+![Notification for changes on multiple tabs](img/Notification-changes-on-multiple-tabs.png)
```

Comment:
talking with @itsptk about this a bit more. I think having this alert always above the tabs makes the most sense. Even if there is only one issue that it's addressing. It seems more sensible. 
The only change required would be to list them out with the same format
details > flavor

Reply:
Agreed, it seems like even if only one change was made on a particular tab, you'd still be interested in knowing that there are pending changes even if you were now looking at a different tab.

Reply:
+1